specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Offer.json

```diff
@@ -0,0 +1,193 @@
+                    },
+                    {
+                        "$ref": "Subscriptions.json#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/Offer"
+                        }
+                    }
+                }
+            }
+        },
+        "/offers": {
+            "get": {
+                "x-ms-examples": {
+                    "Get the list of public offers for the root provider.": {
+                        "$ref": "./examples/Offer/List.json"
+                    }
+                },
+                "description": "Get the list of public offers for the root provider.",
+                "tags": [
+                    "Offers"
+                ],
+                "operationId": "Offers_List",
+                "parameters": [
+                    {
+                        "$ref": "Subscriptions.json#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/OfferList"
+                        }
+                    }
+                },
+                "x-ms-pageable": {
+                    "nextLinkName": "nextLink"
+                }
+            }
+        }   
+    },
+    "definitions": {
+        "Offer": {
+            "description": "Represents an offering of services against which a subscription can be created.",
+            "type": "object",
+            "properties": {
+                "displayName": {
+                    "description": "Display name of offer.",
+                    "type": "string"
+                },
+                "description": {
+                    "description": "Description of offer.",
+                    "type": "string"
+                },
+                "name": {
+                    "description": "The name of the offer.",
+                    "type": "string"
+                },
+                "id": {
+                    "description": "The offer ID",
+                    "type": "string"
+                }
+            }
+        },
+        "OfferList": {
+            "description": "List of public offers.",
+            "type": "object",
+            "properties": {
+                "value": {
+                    "description": "List of public offers.",
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/Offer"
+                    }
+                },
+                "nextLink": {
+                    "type": "string",
+                    "description": "URI to next page."
+                }
+            }
+        }
+    },
```

Comment:
You should add parameters even if empty.

Reply:
Why is that needed? Validators and SDK generation goes fine without it..

Reply:
If you don't have it then you should remove the empty ones from your other specs.  I think it is best to keep all the high level items there.